src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2645,5 +2656,102 @@ private void InterpretCall(int token)
             if (!signature.ReturnType.IsVoid)
                 _stack.Push(callInfo.ReturnValue);
         }
+        
+        private void InterpretNewObj(int token)
+        {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+            MethodSignature signature = method.Signature;
+            TypeDesc owningType = method.OwningType;
+
+            StackItem[] arguments = new StackItem[signature.Length];
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                arguments[i] = PopWithValidation();
+            }
+
+            if (owningType.IsArray)
+            {
+                LowLevelList<int> lengths = default;
+                LowLevelList<int> lowerBounds = default;
+
+                ArrayType arrayType = owningType as ArrayType;
+                Debug.Assert(arrayType != null);
+                
+                if (signature.Length == arrayType.Rank)
+                {
+                    lengths = new LowLevelList<int>();
```

Comment:
These do not need to be growable Lists. The precise size is know at this point.

Reply:
Using a List simplifies the algorithm, I now specify the capacity